napi_error_handling.md

```diff
@@ -0,0 +1,115 @@
+# Error handling
+
+N-API uses both return values and exceptions for error handling.  The sections
```

Comment:
I'll edit to say JavaScript exceptions specifically.  Hopefully as we are discussing a C API that will be enough to remove any potential confusion.